observation/interferometric_pointing.py

```diff
@@ -57,48 +57,49 @@ class NoTargetsUpError(Exception):
-                if not session.target_visible(target, duration=compound_duration):
-                    continue
-                session.label('interferometric_pointing')
-                session.track(target, duration=opts.track_duration, announce=False)
-                for direction in {'x', 'y'}:
-                    for offset in np.linspace(-opts.max_extent, opts.max_extent,
-                                              opts.number_of_steps // 2):
-                        if direction == 'x':
-                            offset_target = [offset, 0.0]
-                        else:
-                            offset_target = [0.0, offset]
-                        user_logger.info("Initiating %g-second track on target '%s'",
-                                         opts.track_duration, target.name)
-                        user_logger.info("Offset of (%f, %f) degrees", *offset_target)
-                        if not kat.dry_run:
-                            session.ants.req.offset_fixed(offset_target[0], offset_target[1], opts.projection)
-                        nd_params = session.nd_params
-                        #session.fire_noise_diode(announce=True, **nd_params)
-                        target.tags = target.tags[:1]  # this is to avoid overloading the cal pipeline
-                        session.track(target, duration=opts.track_duration, announce=False)
-                targets_observed.append(target.name)
-                if opts.max_duration is not None and (time.time() - start_time >= opts.max_duration):
-                    user_logger.warning("Maximum duration of %g seconds has elapsed - stopping script",
-                                        opts.max_duration)
-                    keep_going = False
-                    break
+        try: # Keep going until the time is up or interrupted
+            keep_going = True
+            while keep_going:
+                keep_going = (opts.max_duration is not None)
+                targets_before_loop = len(targets_observed)
+                # Iterate through source list, picking the next one that is up
+                for target in observation_sources.iterfilter(el_limit_deg=opts.horizon):
+                    target.tags = target.tags[:1]  # this is to avoid overloading the cal pipeline
+                    # Check if all offset pointings in compound scan will be up
+                    compound_steps = 1 + 2 * (opts.number_of_steps // 2)
+                    # Add some extra time for slews between pointings
+                    step_duration = opts.track_duration + 4.
+                    compound_duration = compound_steps * step_duration
+                    if not session.target_visible(target, duration=compound_duration):
+                        continue
+                    session.label('interferometric_pointing')
+                    session.track(target, duration=opts.track_duration, announce=False)
+                    for direction in {'x', 'y'}:
```

Comment:
When I originally wrote the script, I put in the cross pattern as a simple way and hoped that somebody would suggest a better scanning pattern. eg. hexagonal close-packed etc. and the off sets would then be put into offset_target as a list. do you have any ideas on a better pattern.

Reply:
My gut feel suggests a circular pattern along the half-power contour, roughly estimated by 7 hexagonal points like so:
```
  o o
 o o o
  o o
```